martha/martha_v3.js

```diff
@@ -527,7 +532,26 @@ async function retrieveFromServers(params) {
             const httpsAccessUrl = generateAccessUrl(drsType, accessId);
             const accessTokenAuth = `Bearer ${accessToken}`;
             console.log(`Requesting DRS access URL for '${url}' from '${httpsAccessUrl}'`);
-            accessUrl = await apiAdapter.getJsonFrom(httpsAccessUrl, accessTokenAuth);
+            /*
+            TODO: Remove this special error obfuscation after confirming that access URL generation is able to handle
+                the load.
+
+            For now, timeout if URL signing takes too long (including WA-90 retries) then try to return the rest of the
+            result in the remaining short GCF lifecycle.
+
+            NOTE: The call to getJsonFrom will not be canceled, and will keep pointlessly WA-90 retrying. But the
+            upstream client to martha_v3 might get a response if there's time left.
+             */
+            // noinspection JSCheckFunctionSignatures
+            const accessUrlAttempt = await Promise.race([
+                apiAdapter.getJsonFrom(httpsAccessUrl, accessTokenAuth),
+                sleep(REQUEST_TIMEOUT_MILLIS, REQUEST_TIMEOUT_RESULT),
+            ]);
```

Comment:
To do: if there's _any_ error returned, even not a timeout, then keep going and return `accessUrl: null`

Reply:
Agreed. Once we have that, I believe this is good to go.

Reply:
Updated and added a test.